From d67ecfb4497148eb43784713d80e268e67cc60db Mon Sep 17 00:00:00 2001
From: Naushir Patuck <naush@raspberrypi.com>
Date: Tue, 30 Nov 2021 10:39:41 +0000
Subject: [PATCH 0321/1085] drivers: bcm2835_unicam: Add logging message when a
 frame is dropped.

If a dummy buffer is still active on a frame start, it indicates that this frame
will be dropped. The explicit logging helps users identify performance issues.

Signed-off-by: Naushir Patuck <naush@raspberrypi.com>
---
 drivers/media/platform/bcm2835/bcm2835-unicam.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/media/platform/bcm2835/bcm2835-unicam.c
+++ b/drivers/media/platform/bcm2835/bcm2835-unicam.c
@@ -951,6 +951,9 @@ static irqreturn_t unicam_isr(int irq, v
 			if (unicam->node[i].cur_frm)
 				unicam->node[i].cur_frm->vb.vb2_buf.timestamp =
 								ts;
+			else
+				unicam_dbg(2, unicam, "ISR: [%d] Dropping frame, buffer not available at FS\n",
+					   i);
 			/*
 			 * Set the next frame output to go to a dummy frame
 			 * if we have not managed to obtain another frame
